src/server.ts

```diff
@@ -85,6 +94,57 @@ export class RecordReplayServer {
         res.statusCode = 500;
         res.end();
       }
+    };
+
+    const httpServer = http.createServer(handler);
+    let httpsServer: https.Server | null = null;
+    if (options.httpsKey && options.httpsCert) {
+      const httpsOptions = {
+        ca: options.httpsCA ? fs.readFileSync(options.httpsCA) : undefined,
+        key: fs.readFileSync(options.httpsKey),
+        cert: fs.readFileSync(options.httpsCert),
+      };
+      httpsServer = https.createServer(httpsOptions, handler);
+    }
+
+    // Copied from https://stackoverflow.com/a/42019773/16286019
+    this.server = net.createServer((socket) => {
+      socket.once("data", (buffer) => {
+        // Pause the socket
+        socket.pause();
+
+        // Determine if this is an HTTP(s) request
+        const byte = buffer[0];
+
+        let server;
+        // First byte of HTTPS is a 22.
+        if (byte === 22) {
+          server = httpsServer;
+        } else if (32 < byte && byte < 127) {
+          server = httpServer;
+        } else {
+          console.error(
+            chalk.red(
+              `Unexpected starting byte of incoming request: ${byte}. Dropping request.`
+            )
+          );
+        }
+
+        if (server) {
+          // Push the buffer back onto the front of the data stream
+          socket.unshift(buffer);
+
+          // Emit the socket to the HTTP(s) server
+          server.emit("connection", socket);
+        }
```

Comment:
suggestion: else case here probably should log out saying we received a HTTP(s) request but the server was not initialized.